src/gfn/containers/trajectories.py

```diff
@@ -213,11 +233,76 @@ def extend(self, other: Trajectories) -> None:
 
         if self._log_rewards is not None and other._log_rewards is not None:
             self._log_rewards = torch.cat(
-                (self._log_rewards, other._log_rewards), dim=0
+                (self._log_rewards, other._log_rewards),
+                dim=0,
             )
         else:
             self._log_rewards = None
 
+        # Either set, or append, estimator outputs if they exist in the submitted
+        # trajectory.
+        if self.estimator_outputs is None and is_tensor(other.estimator_outputs):
+            self.estimator_outputs = other.estimator_outputs
```

Comment:
I think the idea is to be able to extend an empty `Trajectories` instance, say with a stored buffer.

I agree it is dangerous but I think we should support this behaviour.

Admittedly it has been some time since I looked at this so I might be forgetting something.

Reply:
Fair enough!